extensions/amp-jwplayer/1.0/component.js

```diff
@@ -0,0 +1,312 @@
+    onPlayingState,
+    playerId,
+    playlistId,
+    queryParams,
+    ...rest
+  } = props;
+  const playerStateRef = useRef({});
+  const src = useMemo(() => {
+    return makeJwplayerIframeSrc(
+      playerId,
+      playlistId,
+      mediaId,
+      contentSearch,
+      contentRecency,
+      contentBackfill,
+      queryParams,
+      consentParams
+    );
+  }, [
+    playerId,
+    playlistId,
+    mediaId,
+    contentSearch,
+    contentRecency,
+    contentBackfill,
+    queryParams,
+    consentParams,
+  ]);
+
+  const onMessage = useCallback(
+    (e) =>
+      onMessage_(e, {
+        config,
+        adCustParams,
+        adMacros,
+        onPlayingState,
+        onLoad,
+        playerStateRef,
+      }),
+    [config, adCustParams, adMacros, onPlayingState, onLoad]
+  );
+
+  const makeMethodMessage = useCallback(
+    (method) => makeMessage_(method, {autoplay}),
+    [autoplay]
+  );
+  const makeFullscreenMessage = useMemo(
+    () =>
+      !hasFullScreenApi(window)
+        ? () =>
+            JSON.stringify({
+              method: 'setFullscreen',
+              optParams: true,
+            })
+        : null,
+    []
+  );
+
+  if (!checkProps(props)) {
+    return false;
+  }
+
+  return (
+    <VideoIframe
+      ref={ref}
+      playerStateRef={playerStateRef}
+      src={src}
+      makeMethodMessage={makeMethodMessage}
+      makeFullscreenMessage={makeFullscreenMessage}
+      onMessage={onMessage}
+      origin={JWPLAYER_ORIGIN}
+      controls
+      {...rest}
+    />
+  );
+}
+
+const BentoJwplayer = forwardRef(BentoJwplayerWithRef);
+BentoJwplayer.displayName = 'BentoJwPlayer'; // Make findable for tests.
+export {BentoJwplayer};
+
+/**
+ * Makes message to handle different media methods
+ * @param {string} apiMethod
+ * @param {*} options
+ * @return {string} message to post to video iframe
+ */
+function makeMessage_(apiMethod, options) {
+  let jwplayerMethod, optParams;
+  switch (apiMethod) {
+    case 'play': {
+      const {autoplay} = options;
+      if (autoplay) {
+        optParams = true;
+      }
+      jwplayerMethod = 'play';
+      break;
+    }
+    case 'pause': {
+      jwplayerMethod = 'pause';
+      break;
+    }
+    case 'mute': {
+      jwplayerMethod = 'setMute';
+      optParams = true;
+      break;
+    }
+    case 'unmute': {
+      jwplayerMethod = 'setMute';
+      optParams = false;
+      break;
+    }
+    case 'showControls': {
+      jwplayerMethod = 'setControls';
+      optParams = true;
+      break;
+    }
+    case 'hideControls': {
+      jwplayerMethod = 'setControls';
+      optParams = false;
+      break;
+    }
+  }
+  const message = {method: jwplayerMethod, optParams};
+  return JSON.stringify(message);
+}
+
+/**
+ *
+ * @param {Oject} event
+ * @param {{adCustParams: string, adMacros: Object, config: Object, onLoad: OnLoadCb, onPlayingState: OnPlayingStateCb, playerStateRef: {current: Object}}} options
+ */
+function onMessage_(event, options) {
```

Comment:
```suggestion
function onMessage(event, options) {
```

Reply:
it would have the same name as the the callback on line 71 otherwise.